File: ffmpeg-android/src/main/java/com/eugene_lutz/ffmpeg_android/ExecuteResult.java
package com.eugene_lutz.ffmpeg_android;

public final class ExecuteResult
{
	private boolean succeeded;
	private int code;
	private String message;

	public ExecuteResult()
	{
		reset();
	}

	public void reset()
	{
		succeeded = true;
		code = 0;
		message = null;
	}

	public boolean isSucceeded()
	{
		return succeeded;
	}

	public int getCode()
	{
		return code;
	}

	public String getMessage()
	{
		return message;
	}
}

File: ffmpeg-android/src/main/cpp/ExecuteResult.cpp
#include "ExecuteResult.hpp"

ExecuteResult::ExecuteResult(jobject object, JNIEnv* environment) : JavaClass(object, environment)
{
    if (!obj) {
        succeededField = nullptr;
        codeField = nullptr;
        messageField = nullptr;
        return;
    }

    succeededField = GetBooleanField("succeeded");
    codeField = GetIntField("code");
    messageField = GetStringField("message");
}

void ExecuteResult::SetSucceeded(bool succeeded)
{
    if (!obj) {
        return;
    }

    env->SetBooleanField(obj, succeededField, tojboolean(succeeded));
}

void ExecuteResult::SetCode(int code)
{
    if (!obj) {
        return;
    }

    env->SetIntField(obj, codeField, tojint(code));
}

void ExecuteResult::SetMessage(const char* message)
{
    if (!obj) {
        return;
    }

    auto jmessage = env->NewStringUTF(message);
    env->SetObjectField(obj, messageField, jmessage);
}